client/state/selectors/get-media.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Returns media for a specified site ID and query.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @param  {Object}  query  Query object
+ * @return {?Array}         Media
+ */
+export default function getMedia( state, siteId, query ) {
+	const queries = state.media.queries[ siteId ];
+
+	if ( ! queries ) {
+		return null;
+	}
+
+	return queries.getItems( query );
```

Comment:
When it comes down to it though, when we store Immutable objects in state, we're storing an instance of its class, so I don't see much of a difference between something like `state.values()` (of an instance of `Immutable.Map`) vs. `state.getItems()` (of an instance of `MediaQueryManager`). You do raise a good point that Immutable objects can be converted Immutable <-> JS <-> String whereas QueryManager instances can only be converted QueryManager <-> String, though in practice we don't (and probably shouldn't) make use of the JavaScript form of an Immutable instance, except perhaps when initializing from an object.

>Would it be much of a difficult task to turn stateObject.getItems( query ) into getItems( stateObject, query )?

I'm sure it could be done, but maintaining a valid state object shape and passing the required arguments for each argument is not nearly as easy to use as it is in its class form:

```js
const nextManager = receive( {
	items: {},
	queries: {}
}, {
	itemKey: 'ID'
}, posts, query );

// vs.

const nextManager = new PostQueryManager().receive( posts, query );
```